src/components/forms/RegisterForm.tsx

```diff
@@ -44,39 +39,64 @@ const RegisterForm = (): JSX.Element => {
     }
   }
 
-  const onFormSubmit = (e: React.FormEvent): void => e.preventDefault()
-
   return (
-    <Form onSubmit={onFormSubmit}>
+    <Form onSubmit={handleSubmit(handleRegister)}>
       <Form.Group className='mb-3'>
         <Form.Label>{t('form.userName')}</Form.Label>
         <Form.Control
-          required
+          {...register(
+            'username',
+            {
+              required: { value: true, message: t('error.requiredField') },
+              minLength: { value: minLengthUsername, message: t('error.minLengthError', { count: minLengthUsername }) }
+            })
+          }
           type='text'
-          onChange={onUsernameChange}
         />
+        {(errors.username != null) &&
+          <span className='mt-4 text-danger'>
+            {errors.username.message}
+          </span>}
       </Form.Group>
 
       <Form.Group className='mb-3'>
         <Form.Label>{t('form.emailAddress')}</Form.Label>
         <Form.Control
-          required
+          {...register(
+            'email',
+            {
+              required: { value: true, message: t('error.requiredField') },
+              pattern: { value: /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/, message: t('error.emailError') }
```

Comment:
Risky to bring a validation pattern here, for instance this pattern doesn't support `+` sign would it could be valid.
Let's take it out

Reply:
we need to have an error message on this field because the other fields have error messages and the backend lets test@test.comj through
which is not correct in France

Reply:
if a user has an address with a `+ `and is unable to register at that time, we will deal with the situation.

Reply:
> we need to have an error message on this field because the other fields have error messages 

Why is this a valid argument?


> the backend lets [test@test.comj](mailto:test@test.comj) through

The current regular expression doesn't block `test@test.comj` either so I don't get this argument either?


> if a user has an address with a + and is unable to register at that time, we will deal with the situation.

Why bringing problem in the first place? I don't think this regular expression bring value, only problem. The backend does the validation and will send the email confirmation link. Email regular expressions can be painful, let's not come up with our own way unless it really brings value for us or the end user

Reply:
ok great so I delete it